cmd/add.go

```diff
@@ -45,30 +55,39 @@ func add(cmd *cobra.Command, args []string) error {
 		})
 	}
 
-	// Ask username password from prompt
-	organization, err := lib.Prompt("Okta organization", false)
-	if err != nil {
-		return err
+	// Ask Okta organization details and username if not given in command line arguments
+	if organization == "" {
+		organization, err = lib.Prompt("Okta organization", false)
+		if err != nil {
+			return err
+		}
 	}
 
-	oktaRegion, err := lib.Prompt("Okta region ([us], emea, preview)", false)
-	if err != nil {
-		return err
+	if oktaRegion == "" {
```

Comment:
I think there's some edge cases that are going to happen here. Namely domain can be derived from region+org, and region+org can be derived from domain. You end up getting prompted for values that are implied.

```
$ ./aws-okta add --okta-domain example.okta.com
Okta organization: example?

Okta region ([us], emea, preview): us?
```

Reply:
It might be too much trouble since you'd have to mock out `lib.Prompt` probably, but if you feel like it, a table of tests would be cool. Like "if domain flag is set then expect no region, org, or domain prompt" etc.